### ISSUE TITLE
ugently need help on a very simple conceptual problem

### ISSUE THREAD

**Issue by WilsonFok2020** [facebookresearch/differentiable-robot-model#27]
Hi there,

I am trying to understand why I fail to produce identical qdd when I use the same robot to perform forward dynamics twice.

Here is the test script:

``
    
    
    n_data=100
    device="cpu"
    gt_robot_model = DifferentiableKUKAiiwa(device=device)
    train_data = generate_sine_motion_forward_dynamics_data(
        gt_robot_model, n_data=n_data, dt=1.0 / 250.0, freq=0.1
    )
    train_loader = DataLoader(dataset=train_data, batch_size=100, shuffle=False)
    
    for g in [True, False]:
        for d in [True, False]:
            
            for batch_idx, batch_data in enumerate(train_loader):
                    q, qd, qdd, tau = batch_data
                    
            
                    qdd_pred = gt_robot_model.compute_forward_dynamics(
                        q=q, qd=qd, f=tau, include_gravity=g, use_damping=d
                    )
            
                    print (g, d, torch.sum(torch.abs(qdd_pred - qdd)))

printout:
True True tensor(15084.983)
True False tensor(19104.133)
False True tensor(20023.293)
False False tensor(23508.631)

Please let me know which concepts I have failed to grasp, so that I can read up on them.
Many thanks,

**Comment by WilsonFok2020** [comment]
Hi,

While I agree that when the 4 different combinations are run, 4 different outcomes are estimated. However, what I fail to understand is why none of the 4 (qdd_pred) come even close to the “groundtruth”. I define groundtruth as qdd from “generate_sine_motion_forward_dynamics_data” function. 

Because I fail to understand the reason behind this discrepancy, I totally lost when I studied the official example (https://github.com/facebookresearch/differentiable-robot-model/blob/main/examples/learn_forward_dynamics_iiwa.py)

The example says that we get the groundtruth joint accelerations from the groundtruth robot by calling “generate_sine_motion_forward_dynamics_data” function. Our learnable robot is identical to the groundtruth robot except it has several tunable physical parameters (mass, center of mass and inertia). Well, to calibrate the learnable robot, aren’t we expecting that the tunable parameters (upon convergence) be the same as the groundtruth robot’s mass, center of mass and inertia?

If not so, what are we trying to achieve in the example?
By the way, when I run the example straight from the repository, the learnable robot has not converged. It surprised me; I do not know why.

Thanks for answering my questions.

**Comment by WilsonFok2020** [comment]
Thank you very much.
